src/sif/motorcyclecost.cc

```diff
@@ -307,7 +322,8 @@ class MotorcycleCost : public DynamicCost {
    */
   virtual const NodeFilter GetNodeFilter() const {
     // throw back a lambda that checks the access for this type of costing
-    return [](const baldr::NodeInfo* node) { return !(node->access() & kMotorcycleAccess); };
+    auto access_mask = (ignore_access_ ? kAllAccess : kMotorcycleAccess);
```

Comment:
i dont think we store nodes that don't have any access bits flipped so i think we can just say something like

    !((node->access() & kMotorcycleAccess) || ignore_access)

I think what you have is equivalent though so dont feel obligated to change it. in this case the filter is only created once and used over and over so the branch doesnt really penalize performance